proto/stride/stakeibc/trade_route.proto

```diff
@@ -2,9 +2,35 @@ syntax = "proto3";
 package stride.stakeibc;
 
 import "gogoproto/gogo.proto";
+import "stride/stakeibc/ica_account.proto";
 
 option go_package = "github.com/Stride-Labs/stride/v14/x/stakeibc/types";
 
+// Define new types less general and more specific to this task, old types to be removed in later PR
+
+message TradeHop {
+  string      transfer_channel_id = 1;
+  ICAAccount  from_account = 2 [ (gogoproto.nullable) = true ];
+  ICAAccount  to_account = 3 [ (gogoproto.nullable) = true ];
+}
+
+message TradeRoute {
+	string reward_denom_on_host_zone = 1;
+	string reward_denom_on_reward_zone = 2;
+	string reward_denom_on_trade_zone = 3;
+	string target_denom_on_trade_zone = 4;
+	string target_denom_on_host_zone = 5;
+
+	TradeHop host_to_reward_hop = 6;
+	TradeHop reward_to_trade_hop = 7;
+	TradeHop trade_to_host_hop = 8;
```

Comment:
should we make these nullable is false? I feel like nullable nested structs get kinda messy from protos

For instance, if nullable is false and we have an uninitialized struct, we can do `tradeRoute.host_to_reward_hop.transfer_channel_id` and it will return an empty string. Otherwise, we have to either first check that each field is parent field is not null or use the Getters

Reply:
Or we could define a function to ValidateTradeRoute which checks that everything is correctly initialized and call that in the start of each of the functions which uses a trade route.  I don't think the proto nullable = false actually does much to prevent the non-initialized mistake, it just prevents the runtime error from referring to a field on something which doesn't exist but an empty string is also very bad and would break things downstream

Reply:
yeah agreed on the ValidateTradeRoute thing! I still think it's a bit cleaner to not deal with points though. I always thought that was a pain when we had some on our host zone structs and I noticed other projects (like the client/connection/channel objects in ibc-go) just set nullable to false and it's always easier to deal with

Reply:
would not using pointers solve [this](https://github.com/Stride-Labs/stride/pull/980/files#diff-8f6f0a1d11c2bb7abe4951c1e628218eb98be41b99e1372df42ce56fdc4f917bR71-R75)?